### ISSUE TITLE
Add auto saving

### ISSUE THREAD

**Issue by WiIIiam278** [haroohie-club/SerialLoops#209]
We should have an option (preferably on by default) to automatically save / build the project in the background. A bit of optimisation is needed to only save periodically and not constantly

**Comment by jonko0493** [comment]
Autosaving makes sense to me actually as it requires very little optimization (saving is very cheap). Building is a bit trickier as that actually takes resources. I'd argue that saving is necessary but building is less so, especially since we have the ability to detect unbuilt saves on load.

**Comment by WiIIiam278** [comment]
Sounds good—should still build at least on exit

**Comment by jonko0493** [comment]
Agreed on that; the build on load feature is intended more for crashes and such anyway

**Comment by jonko0493** [comment]
Thinking about making this about auto-building in an asynchronous background task and then checking at Build & Run time if we need to build to speed that up.
